openadmet/models/architecture/chemprop.py

```diff
@@ -22,30 +22,97 @@
 @model_registry.register("ChemPropModel")
 class ChemPropModel(LightningModelBase):
     """
-    ChemProp regression model
+    ChemProp regression model.
+
+    This class implements a ChemProp-based regression model using message passing neural networks (MPNNs)
+    for molecular property prediction. It supports various configurations for message passing, aggregation,
+    and feed-forward network (FFN) layers. Can be initialized from the CheMeleon foundation model [REF], overriding
+    settings for depth, message hidden dim, messages, and aggregation.
+
+    Attributes
+    ----------
+    type : str
+        The type of the model.
+    n_tasks : int
+        Number of prediction tasks.
+    messages : str
+        Type of message passing ("bond" or "atom").
+    aggregation : str
+        Aggregation method ("mean" or "norm").
+    depth : int
+        Number of message passing steps.
+    message_hidden_dim : int
+        Hidden dimension size for message passing.
+    ffn_hidden_dim : int
+        Hidden dimension size for the feed-forward network.
+    ffn_num_layers : int
+        Number of layers in the feed-forward network.
+    scaler : object
+        Scaler used for target normalization.
+    normalized_targets : bool
+        Whether targets are normalized.
+    batch_norm : bool
+        Whether to use batch normalization.
+    dropout : float
+        Dropout rate.
+    from_chemeleon : bool
+        Whether to use the CheMeleon foundation model.
+    monitor_metric : str
+        The metric to monitor during training.
+    metric_list : list
+        List of metrics to use for evaluation.
+    warmup_epochs : int
+        Number of warmup epochs for learning rate scheduling.
+    init_lr : float
+        Initial learning rate.
+    max_lr : float
+        Maximum learning rate.
+    final_lr : float
+        Final learning rate.
+
     """
 
     type: ClassVar[str] = "ChemPropModel"
-    batch_norm: bool = False
-    monitor_metric: str = "val_loss"
-    metric_list: list = ["mse", "mae", "rmse"]
-    mod_params: dict = {}
-    from_chemeleon: bool = False
+    n_tasks: int = 1
+    messages: str = "bond"
+    aggregation: str = "norm"
     depth: int = 3
     message_hidden_dim: int = 300
     ffn_hidden_dim: int = 300
     ffn_num_layers: int = 1
-    messages: str = "bond"
-    aggregation: str = "norm"
+    scaler: object = None
```

Comment:
Scaler now as a class attribute seems error prone given previous issues (?) we had with it. I can't remember exactly why this wasn't possible before but IIRC it was because the correct scaler is not available on `__init__`  as dependent on data.

At a minimum this will need changes in anvil + testing.